aspnet/security/authorization/secure-data.rst

```diff
@@ -0,0 +1,369 @@
+
+Require SSL and authenticated users
+------------------------------------
+
+In the ``ConfigureServices`` method of the *Startup.cs* file, add the :dn:cls:`~Microsoft.AspNetCore.Mvc.RequireHttpsAttribute` authorization filter that requires all requests use HTTPS:
+
+.. literalinclude::  secure-data/samples/final/Startup.cs
+  :language: c#
+  :start-after:  #region snippet_SSL
+  :end-before:  #endregion
+  :dedent: 12
+
+If you are using Visual Studio, see :ref:`Enable-ssl-visual-studio-label`.
+
+Set the default authentication policy to require users to be authenticated. You can opt out of authentication at the controller or action method with the ``[AllowAnonymous]`` attribute. With this approach, any new controllers added will automatically require authentication, which is more fail safe than relying on new controllers to include the ``[Authorize]`` attribute. Add the following to  the ``ConfigureServices`` method of the *Startup.cs* file:
+
+.. literalinclude::  secure-data/samples/final/Startup.cs
+  :language: c#
+  :start-after: #region snippet_defaultPolicy
+  :end-before:  #endregion
+  :dedent: 12
+
+Add ``[AllowAnonymous]`` to the home controller so anonymous users can get information about the site before they register.
+
+Configure the test account
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The ``SeedData`` class creates a test user account. Use the  :doc:`Secret Manager tool </security/app-secrets>` to set a password for the account. Do this from the project directory (the directory containing *Program.cs*).
+
+.. code-block:: none
+
+  dotnet user-secrets set SeedUserPW <PW>
+
+Update ``Configure`` to use the test password:
+
+.. literalinclude::  secure-data/samples/final/Startup.cs
+  :language: c#
+  :start-after: #region snippet_SeedUserPW
+  :end-before: #endregion
+  :dedent: 12
+
+Add the test accounts user ID to the seed data. Only one contact is shown, add the user ID to all contacts:
+
+.. literalinclude::  secure-data/samples/final/Data/SeedData.cs
+  :language: c#
+  :start-after: #region snippet_Contact
+  :end-before: #endregion
+  :dedent: 12
+  :emphasize-lines: 9
+
+Delete all the records in the ``Contact`` table and restart the app to seed the database. If you're using Visual Studio you might need to stop IIS Express to force the seed initializer to run.
+
+Resource based authorization
+-----------------------------
+
+- Create an *Authorization* folder for the handlers and classes we will create to implement authorization.
+- Create a *ContactIsOwnerAuthorizationHandler* class we can invoke to verify the user acting on the resource owns the resource. Create this in the *Authorization* folder.
+
+.. literalinclude:: secure-data/samples/final/Authorization/ContactIsOwnerAuthorizationHandler.cs
+  :language: c#
+
+The ``ContactIsOwnerAuthorizationHandler`` returns ``Succeed`` if the user is the contact owner. We're not checking the requirement parameter; an owner can perform any requirement on data they own.
```

Comment:
The ``ContactIsOwnerAuthorizationHandler`` calls ``context.Succeed`` if the current authenticated user is the contact owner. As we allow contact owners to perform any operation on their own data we don't need to check the operation passed in the requirement parameter,